syncer/syncer_test.go

```diff
@@ -135,6 +135,32 @@ func (s *testSyncerSuite) resetMaster() {
 	s.db.Exec("reset master")
 }
 
+func (s *testSyncerSuite) catchUpBinlog() {
+	ch := make(chan interface{})
+	ctx, cancel := context.WithCancel(context.Background())
+
+	go func() {
+		for {
+			ev, _ := s.streamer.GetEvent(ctx)
+			if ev == nil {
+				return
+			}
+			ch <- struct{}{}
+		}
+	}()
+
+	for {
+		t := time.NewTimer(10 * time.Millisecond)
+		select {
+		case <-ch:
+			t.Stop()
```

Comment:
whether needing to call `cancel` here? and whether needing to *wait* the above goroutine to return?